### PR TITLE
Add request to Edit Handler

### PR DESCRIPTION
**Pull request by rajeev**
Passing in the request adds significant flexibility when developing new panels. This was developed to support adding multiple images to a Page model by drag and drop.

### CODE REVIEW

File: wagtail/admin/edit_handlers.py [hunk 1]
```diff
@@ -153,6 +153,10 @@ def bind_to_instance(self, instance=None, form=None):
             raise ValueError("EditHandler did not receive a form object")
         new.form = form
 
+        if not request:
```
Comment (BertrandBordage) [inline wagtail/admin/edit_handlers.py]: Please use the more obvious `request is None` instead.

File: wagtail/admin/tests/test_edit_handlers.py [hunk 2]
```diff
@@ -2,9 +2,11 @@
 
 import mock
 from django import forms
+from django.contrib.auth.models import AnonymousUser
 from django.core import checks
 from django.core.exceptions import FieldDoesNotExist, ImproperlyConfigured
-from django.test import TestCase, override_settings
+from django.test import RequestFactory, TestCase, override_settings
+
```
Comment (BertrandBordage) [inline wagtail/admin/tests/test_edit_handlers.py]: Please remove this extra newline.

File: wagtail/admin/views/pages.py [hunk 3]
```diff
@@ -310,6 +311,7 @@ def edit(request, page_id):
     edit_handler = page_class.get_edit_handler()
     form_class = edit_handler.get_form_class()
 
+
```
Comment (BertrandBordage) [inline wagtail/admin/views/pages.py]: Please remove this extra newline, according to PEP8 we shouldn’t have two successive newlines in a function or method.

**Comment by rajeev** [comment]
Will make these changes asap and update the PR

**Comment by SalahAdDin** [comment]
:+1:

**Comment by thibaudcolas** [comment]
🎉 merged! This will make its way to a Wagtail site near you in time for Wagtail 2.1. Thank you @rajeev.

**Pull request merged**